OpenSim/Region/CoreModules/ServiceConnectors/Interregion/RESTInterregionComms.cs

```diff
@@ -91,9 +91,7 @@ public virtual void Initialize(Scene scene, IConfigSource config)
 
 
                 IConfig startupConfig = config.Configs["Communications"];
-                if ((startupConfig == null)
-                    || (startupConfig != null)
-                    && (startupConfig.GetString("InterregionComms", "RESTComms") == "RESTComms"))
+                if (startupConfig != null && startupConfig.GetString("InterregionComms", "RESTComms") == "RESTComms")
                 {
```

Comment:
I agree with that aside. Yet another TODO list item :)

On Wed, Jul 20, 2016 at 1:15 PM, Jim Tarber <notifications@github.com>
wrote:

> In
> OpenSim/Region/CoreModules/ServiceConnectors/Interregion/RESTInterregionComms.cs
> <https://github.com/InWorldz/halcyon/pull/99#discussion_r71598292>:
>
> > @@ -91,9 +91,7 @@ public virtual void Initialize(Scene scene, IConfigSource config)
> >
> >
> >                  IConfig startupConfig = config.Configs["Communications"];
> > -                if ((startupConfig == null)
> > -                    || (startupConfig != null)
> > -                    && (startupConfig.GetString("InterregionComms", "RESTComms") == "RESTComms"))
> > +                if (startupConfig != null && startupConfig.GetString("InterregionComms", "RESTComms") == "RESTComms")
> >                  {
>
> while the second startupConfig != null test was previously redundant
> (because of the || ), this commit changes the semantics for if there's no
> Communications section at all, from defaulting to using this module to not
> using it. That seems like it is the improved behavior; I'm just noting the
> semantic change (which should be okay here -- I don't think we'll have
> regions that *don't* have a Communications section and *do* want
> RESTComms enabled. As a bit of an aside though, I do wish code like this
> checking for specific names was case-insensitive.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/InWorldz/halcyon/pull/99/files/d332cd57f263cafbc7e1f3c1a6240845e231d199#r71598292>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABGoaCNtqAP98EZx2u9MzoZ-wsWrMUUTks5qXoH9gaJpZM4G0WVe>
> .
>